fb/app/models/user.rb

```diff
@@ -57,33 +46,34 @@ def friends?(user)
   end
 
   def recommended_friends
-    friends = User.all.map { |user| user if (!friends? user) && (mutual_friends(user).size > 1) && user != self }.compact
+    friends = User.all.map { |user| user if (!self.friends? user) && (mutual_friends(user).size > 1) }.compact
     if friends.size < 10
-      ((friends + unknown_users(self)) - [self]) - inverse_pending_friends
+      ((friends + unknown_users(self)) - [self]) - pending_requests
     else
-      (friends - [self]) - inverse_pending_friends
+      (friends - [self]) - pending_requests
     end
   end
 
   def confirmed_friends
-    friendships.map { |friendship| friendship.friend if friendship.confirmed }.compact
+    confirmed_friendships.map { |friendship| friendship.friend }
```

Comment:
- [ ] We can use instead use `has_many :confirmed_friends, through: :confirmed_friendships`. This association will also give you a number of helper method like `confirmed_friend_ids` and many other. Please take a look at [this](https://api.rubyonrails.org/classes/ActiveRecord/Associations/ClassMethods.html) under "collection associations" section to check other generated methods from this association.

Reply:
Thank you for providing the resource, it was very helpful